packages/nimble-components/src/chip/specs/README.md

```diff
@@ -0,0 +1,200 @@
+- Implementing a "chip container" component that would be responsible for laying out the chips in a particular fashion, as well as possibly managing their removal if the "remove" button on a chip is pressed. While this may be useful at some point it is not necessary for the `Chip` component to have value.
+
+### Features
+
+- Optionally removable
+    - Note that the `Chip` won't remove itself, and it will be up to the client to remove it from the DOM.
+- Allows slotted content to be placed to left of text content. Slotted content will be styled automatically to be appropriately sized for the chip (i.e. given a height of `controlSlimHeight`). Types of slotted content that will be documented as acceptible are:
+    - icons
+    - buttons with `content-hidden` set and `appearance` set to `ghost`.
+- If a width is set and the label content is larger than the bounds of its container it will present with an ellipsis, and a tooltip will show the entire contents.
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+SLE keywords display:
+
+![](./spec-images/SLE-keywords.PNG)
+
+SLE axis selection:
+
+![](./spec-images/SLE-axis-selection.PNG)
+
+Nigel chat attachment chip:
+
+![](./spec-images/Nigel-attachment-chip.PNG)
+
+[Infragistics Chip component](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/inputs/chip)
+
+---
+
+## Design
+
+### API
+
+_The key elements of the component's public API surface:_
+
+- _Component Name_: `nimble-chip`
+- _Props/Attrs_:
+    - `prevent-remove` - set to hide the remove button
+    - `appearance` - supports `outline` and `block` appearances
+    - `disabled`
+- _Methods_
+- _Events_
+    - `remove` - fired when the chip remove button is pressed (or \<Del\> or \<Backspace\> is pressed), and `prevent-remove` hasn't been set.
+- _Slots_
+    - `start` - content placed to the left of the chip text
+    - (default) - for the primary label text
+- _CSS Classes and CSS Custom Properties that affect the component_
+    - not supported
+
+### Examples
+
+```html
+<nimble-chip> "Option Label" </nimble-chip>
+```
+
+![](./spec-images/Chip-normal.PNG)
+
+```html
+<nimble-chip prevent-remove> "Option Label" </nimble-chip>
+```
+
+![](./spec-images/Chip-no-remove.PNG)
+
+```html
+<nimble-chip>
+    <nimble-icon-check slot="start"> </nimble-icon-check>
+    "Option Label"
+</nimble-chip>
+```
+
+![](./spec-images/Chip-checked.PNG)
+
+### Anatomy
+
+```html
+<template aria-disabled="${x => x.disabled}">
+    <div class="control" part="control">
+        ${startSlotTemplate(context, definition)}
+        <span class="content" part="content">
+            <slot></slot>
+        </span>
+        ${endSlotTemplate(context, definition)}
+        ${when(x => !x.preventRemove, html<Chip>`
+            <nimble-button content-hidden @click="${x => x.handleRemoveClick()}">
+                <nimble-icon-times>
+                </nimble-icon-times>
+            </nimble-button>
```

Comment:
nit: This could be improved a little

```suggestion
            <nimble-button content-hidden @click="${x => x.handleRemoveClick()}" title="Remove">
                <nimble-icon-times slot="start">
                </nimble-icon-times>
                Remove
            </nimble-button>
```